packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactDelegate.java

```diff
@@ -237,17 +238,29 @@ public boolean onKeyLongPress(int keyCode) {
 
   public void reload() {
     DevSupportManager devSupportManager = getDevSupportManager();
-    if (devSupportManager != null) {
-      // With Bridgeless enabled, reload in RELEASE mode
-      if (devSupportManager instanceof ReleaseDevSupportManager
-          && ReactFeatureFlags.enableBridgelessArchitecture
-          && mReactHost != null) {
-        // Do not reload the bundle from JS as there is no bundler running in release mode.
-        mReactHost.reload("ReactDelegate.reload()");
+    if (devSupportManager == null) {
+      return;
+    }
+
+    // Reload in RELEASE mode
+    if (devSupportManager instanceof ReleaseDevSupportManager) {
+      // Do not reload the bundle from JS as there is no bundler running in release mode.
+      if (ReactFeatureFlags.enableBridgelessArchitecture) {
+        if (mReactHost != null) {
+          mReactHost.reload("ReactDelegate.reload()");
+        }
       } else {
-        devSupportManager.handleReloadJS();
+        UiThreadUtil.runOnUiThread(() -> {
```

Comment:
Is this necessary to run it on UIThread ?

Also, what's the failure we are getting with `devSupportManager.handleReloadJS()` ?

Reply:
> Is this necessary to run it on UIThread ?

`recreateReactContextInBackground` is required to be called from ui thread.

> Also, what's the failure we are getting with `devSupportManager.handleReloadJS()` ?

it seems like we don't have any error handling for handleReloadJS()? 
https://github.com/facebook/react-native/blob/995d5b832d3ce354b908f1d8b138bda824260f91/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/BridgeDevSupportManager.java#L191-L216

Reply:
Isn't this the same thing happening in Bridgeless in Android which is working :

https://github.com/facebook/react-native/blob/78ab5f4b836d64b71760754c7cce1315630640c7/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/BridgelessDevSupportManager.java#L100-L112

Reply:
yes, it's the same. sorry i may misunderstand your original question "what's the failure we are getting with devSupportManager.handleReloadJS()".

i didn't change any code for devSupportManager.handleReloadJS() inside the ReactDelegate.reload(). just reorganized the if/else a little bit and having some early returns without nested if block in depth.
could you explain more about your question?

Reply:
My question was with existing:

[`devSupportManager.handleReloadJS()`](https://github.com/facebook/react-native/blob/993f500fcf845563728e4070040a727e0c79f6d6/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactDelegate.java#L248)

That's correctly defaulting to [BridgeDevSupportManager.handleReloadJS()](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/BridgeDevSupportManager.java#L191-L216) so why does Bridge in RELEASE mode doesn't have reload() working was my doubt.